packages/migrate/src/__tests__/MigrateDiff.test.ts

```diff
@@ -0,0 +1,513 @@
+            `)
+    })
+
+    it('should fail if more than 1 --from-... is provided', async () => {
+      ctx.fixture('empty')
+
+      const result = MigrateDiff.new().parse(['--preview-feature', '--from-empty', '--from-url=file:dev.db'])
+      await expect(result).rejects.toThrowErrorMatchingInlineSnapshot(`
+              2 \`--from-...\` paramater(s) provided. 1 must be provided.
+              0 \`--to-...\` paramater(s) provided. 1 must be provided.
+              See prisma migrate diff -h
+            `)
+    })
+
+    it('should fail if more than 1 --to-... is provided', async () => {
+      ctx.fixture('empty')
+
+      const result = MigrateDiff.new().parse(['--preview-feature', '--to-empty', '--to-url=file:dev.db'])
+      await expect(result).rejects.toThrowErrorMatchingInlineSnapshot(`
+              0 \`--from-...\` paramater(s) provided. 1 must be provided.
+              2 \`--to-...\` paramater(s) provided. 1 must be provided.
+              See prisma migrate diff -h
+            `)
+    })
+
+    it('should fail if schema does no exists, --from-schema-datasource', async () => {
+      ctx.fixture('empty')
+      expect.assertions(2)
+
+      try {
+        await MigrateDiff.new().parse([
+          '--preview-feature',
+          '--from-schema-datasource=./doesnoexists.prisma',
+          '--to-empty',
+        ])
+      } catch (e) {
+        expect(e.code).toEqual(undefined)
+        expect(e.message).toMatchInlineSnapshot(`
+          Could not find a schema.prisma file that is required for this command.
+          You can either provide it with --schema, set it as \`prisma.schema\` in your package.json or put it into the default location ./prisma/schema.prisma https://pris.ly/d/prisma-schema-location
+        `)
+      }
+    })
+
+    it('should fail for empty/empty', async () => {
+      ctx.fixture('empty')
+      expect.assertions(2)
+
+      try {
+        await MigrateDiff.new().parse(['--preview-feature', '--from-empty', '--to-empty'])
+      } catch (e) {
+        expect(e.code).toEqual(undefined)
+        expect(e.message).toMatchInlineSnapshot(`
+          Could not find a schema.prisma file that is required for this command.
+          You can either provide it with --schema, set it as \`prisma.schema\` in your package.json or put it into the default location ./prisma/schema.prisma https://pris.ly/d/prisma-schema-location
+        `)
+      }
+    })
+  })
+
+  describe('sqlite', () => {
+    // TODO next 2 tests: is it expected to not fail when diffing from/to an unexisting sqlite db?
+    it('should diff --from-empty --to-url=file:doesnotexists.db', async () => {
+      ctx.fixture('schema-only-sqlite')
+
+      const result = MigrateDiff.new().parse(['--preview-feature', '--from-empty', '--to-url=file:doesnotexists.db'])
+      await expect(result).resolves.toMatchInlineSnapshot(``)
+      expect(ctx.mocked['console.info'].mock.calls.join('\n')).toMatchInlineSnapshot(``)
```

Comment:
Shouldn't this produce a diff too?

Reply:
Or does it produce a diff only when there's something to compare on both ends?

Reply:
Here it's empty to empty, so nothing

Reply:
Ah sorry, I meant https://github.com/prisma/prisma/pull/11440/files/c4d0402834ee1603f39a05bcb74310edaf569539#diff-06b471162d0b222816569f260465ceefd7e0a609009a4a5c81ca8e83cb218db7R135

Reply:
But there should be an error here on this at least?

Reply:
There yes, I'll check what's going on